### PR TITLE
Fixed an issue where the Avant Gardens Survival buff station was not spawning and implemented its' script

### LINKED ISSUE #341: BUG: Survival Buff Station Doesn't Show Up
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### Issue Description

There is an achievement for building the Survival Buff Station 100 times in Avant Gardens Survival. However, I have never seen any quick builds throughout my entire time playing Survival.

image

### Reproduction steps

Play Avant Gardens Survival and try (and fail) to find any sort of Quick Build.

### Expected Behavior

I don't even remember this thing existing in the original game to be honest. So either the Quick Build should be implemented, or the achievement should be removed.

### Environment

Ubuntu Server

### PR DESCRIPTION
**Pull request by EmosewaMC**
Fixes #341 
This PR fixes 2 issues:
First was the issue of the buff station just not spawning.  This was due to a missing spawner network activation in the base survival server when starting the waves.
The second issue was the buff station just not doing anything.  This was due to a missing script.  This script was added and its behavior based on that located at [explorer.lu](https://explorer.lu/objects/6412/9).  The behavior involves every 2.5 seconds, [to heal, buff and imagine all players in its radius by 1.  ](https://explorer.lu/skills/behaviors/1784)
Tested on Ubuntu and had zero issues getting achievement progress and buff station is functioning as expected.

### CODE REVIEW

**Comment by EmosewaMC** [comment]
@Wincent01 @MickVermeulen I'm not sure who worked on the quick build stuff but if one of you two could give this a look or point me to someone who can let me know the following that'd be great!

The buff station is supposed to smash after 10 seconds, however in my testing this was not the case.  The buff station would stay up until it is either manually smashed or 99 cycles pass.  I have put a redundant smash timer in the script to smash the object after the 10 seconds to counter this behavior.  
On [this](https://explorer.lu/objects/6412/48) page, the `time_before_smash` is 10 so thats where that value comes from.  
@Xiphoseer The skill and by extension the behavior attached to this Entity is 201 (BehaviorId #1784) which heals, imagines and buffs for 1 every 2.5 seconds to all players in its radius as opposed to spawning powerups.  Maybe this was changed at some point in live to spawn powerups instead but this is the behavior I have implemented for now based on the information provided.

File: dScripts/AgSurvivalBuffStation.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include "dLogger.h"
+
+void AgSurvivalBuffStation::OnRebuildComplete(Entity* self, Entity* target) {
+    auto skillComponent = self->GetComponent<SkillComponent>();
+
+    if (skillComponent == nullptr) return;
+
+    skillComponent->CalculateBehavior(201, 1784, self->GetObjectID());
+    
+    self->AddTimer("DestroyAfter10Seconds", 10.0f);
```
Comment (MickVermeulen) [inline dScripts/AgSurvivalBuffStation.cpp:12]: You can also use a callback timer here

Comment (MickVermeulen) [reply dScripts/AgSurvivalBuffStation.cpp:12]: (Non blocking)

Comment (EmosewaMC) [reply dScripts/AgSurvivalBuffStation.cpp]: Changed this out for a callback timer.  I dont understand _how_ exactly they work yet but I followed what a callback timer looks like for other scripts and that seemed to work.

**Comment by EmosewaMC** [comment]
@averysumner Jett mentioned I can ask you about Quick Builds so im pinging you on the issue for your thoughts on the above if you have any.  Appreciate it!

**Comment by averysumner** [comment]
Looks good to me.

**Comment by DarwinAnim8or** [comment]
Looked over by two DLU members, merging

**Pull request merged**